packages/calypso-e2e/src/lib/utils/metrics.ts

```diff
@@ -0,0 +1,234 @@
+/**
+ * External dependencies
+ */
+import type { Page, Browser } from 'playwright';
+
+type EventType =
+	| 'click'
+	| 'focus'
+	| 'focusin'
+	| 'keydown'
+	| 'keypress'
+	| 'keyup'
+	| 'mouseout'
+	| 'mouseover';
+
+interface TraceEvent {
+	cat: string;
+	name: string;
+	dur?: number;
+	args: {
+		data?: {
+			type: EventType;
+		};
+	};
+}
+
+interface Trace {
+	traceEvents: TraceEvent[];
+}
+
+interface WebVitalsMeasurements {
+	CLS?: number;
+	FCP?: number;
+	FID?: number;
+	INP?: number;
+	LCP?: number;
+	TTFB?: number;
+}
+
+/**
+ * Metrics class to collect performance metrics.
+ */
+export class Metrics {
```

Comment:
Is there a good reason why we're not using `@wordpress/e2e-test-utils-playwright` here?

Reply:
I tried quickly, it was not straightforward because of import related issues. I'm not very familiar with the e2e test setup of calypso to be able to solve this properly.

Reply:
Will take a stab at it

Reply:
It was pretty straightforward, but we also needed to add a peer dependency and shuffle some things around, see 47f9e94c3bc4e5dc4d99b1bc91af3ff00e51309a

Reply:
Thanks :)

Reply:
I think there's a small risk that the class `Metrics` doesn't work as expected because of the iframe in some situations though.

Reply:
I see. If it happens, it might make sense to extend it rather than redefine it though?

Reply:
Happy to start that way and iterate.